config-model-generator/src/main/java/io/strimzi/build/kafka/metadata/KafkaConfigModelGenerator.java

```diff
@@ -21,52 +16,37 @@
 import org.apache.kafka.raft.RaftConfig;
 import scala.collection.Iterator;
 
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.security.AccessController;
 import java.security.PrivilegedAction;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Properties;
 import java.util.TreeMap;
-import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-@SuppressWarnings("unchecked")
-public class KafkaConfigModelGenerator {
-
-    public static void main(String[] args) throws Exception {
-
-        String version = kafkaVersion();
-        Map<String, ConfigModel> configs = configs();
-        ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).enable(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY);
-        ConfigModels root = new ConfigModels();
-        root.setVersion(version);
-        root.setConfigs(configs);
-        mapper.writeValue(new File(args[0]), root);
-    }
+import static java.security.AccessController.doPrivileged;
 
-    private static String kafkaVersion() throws IOException {
-        Properties p = new Properties();
-        try (InputStream resourceAsStream = Kafka.class.getClassLoader().getResourceAsStream("kafka/kafka-version.properties")) {
-            p.load(resourceAsStream);
-        }
-        return p.getProperty("version");
-    }
+@SuppressWarnings("unchecked")
+public class KafkaConfigModelGenerator extends CommonConfigModelGenerator {
 
-    private static Map<String, ConfigModel> configs() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+    @Override
+    protected Map<String, ConfigModel> configs() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
         ConfigDef def = brokerConfigs();
         Map<String, String> dynamicUpdates = brokerDynamicUpdates();
         Method getConfigValueMethod = def.getClass().getDeclaredMethod("getConfigValue", ConfigDef.ConfigKey.class, String.class);
-        getConfigValueMethod.setAccessible(true);
+        doPrivileged((PrivilegedAction) () -> {
```

Comment:
I don't think we need the whole `doPrivileged` thing, since the config model generator operates at build time => We're in control of whether there's a security manager.

Reply:
spotbugs is complaining about this. Should I suppress the warning?

Reply:
Yes, I would.